server/impl/meetup.go

```diff
@@ -1,15 +1,762 @@
+			Message: "Specified meetup not found.",
+		})
+	} else if err != nil {
+		logger.WithError(err).Error("Could not access meetup DB")
+		return InternalServerError{}
+	}
+
+	if dbMeetup.Cancelled == true {
+		return operations.NewGetMeetupIDBadRequest().WithPayload(&models.Error{
+			Code:    http.StatusBadRequest,
+			Message: "This meetup has been cancelled",
+		})
+	}
+
+	if err = tx.Model(&dbMeetup).Association("Tags").Find(&dbMeetup.Tags); err != nil {
+		logger.WithError(err).Error("Unable to find user tags")
+		return InternalServerError{}
+	}
+
+	if err = tx.Model(&dbMeetup).Association("Attendees").Find(&dbMeetup.Attendees); err != nil {
+		logger.WithError(err).Error("Unable to find meetup attendee information")
+		return InternalServerError{}
+	}
+
+	var idStr string
+	if id := SessionFromContext(ctx).Values[UserID]; id == nil {
+		idStr = ""
+	} else {
+		if _, err := db.UserIDFromString(idStr); err != nil {
+			logger.Error("Session has invalid user ID")
+			return InternalServerError{}
+		}
+		idStr = id.(string)
+	}
+
+	return operations.NewGetMeetupIDOK().WithPayload(dbMeetupToModelMeetup(&dbMeetup, idStr))
+}
+
+func (i *Implementation) PostMeetup(params operations.PostMeetupParams, _ interface{}) middleware.Responder {
+	ctx := params.HTTPRequest.Context()
+	logger := log.WithContext(ctx)
+
+	var dbMeetup db.Meetup
+	id := SessionFromContext(ctx).Values[UserID]
+	if _, err := db.UserIDFromString(id.(string)); err != nil {
+		logger.Error("Session has invalid user ID")
+		return InternalServerError{}
+	}
+
+	modelMeetup := modelMeetupRequestBodyToModelMeetup(params.Meetup, id.(string))
+	if err := i.modelMeetupToDBMeetup(&dbMeetup, &modelMeetup); err != nil {
+		logger.WithError(err).Error("Failed to create db meetup object")
+		return InternalServerError{}
+	}
+
+	tx := i.DB().WithContext(ctx)
+	if err := tx.Create(&dbMeetup).Error; err != nil {
+		logger.WithError(err).Error("Failed to create meetup")
+		return InternalServerError{}
+	}
+
+	if err := i.insertMeetupTagsIntoDB(ctx, &dbMeetup, &modelMeetup); err != nil {
+		logger.WithError(err).Error("Failed to insert meetup tags")
+		return InternalServerError{}
+	}
+
+	return operations.NewPostMeetupCreated().WithPayload(dbMeetupToModelMeetup(&dbMeetup, id.(string)))
+}
+
+func (i *Implementation) PatchMeetupID(params operations.PatchMeetupIDParams, _ interface{}) middleware.Responder {
+	ctx := params.HTTPRequest.Context()
+	logger := log.WithContext(ctx)
+
+	tx := i.DB().WithContext(ctx)
+	var dbMeetup db.Meetup
+	err := tx.First(&dbMeetup, params.ID).Error
+	if errors.Is(err, gorm.ErrRecordNotFound) {
+		return operations.NewPatchMeetupIDNotFound().WithPayload(&models.Error{
+			Code:    http.StatusNotFound,
+			Message: "Specified meetup not found.",
+		})
+	} else if err != nil {
+		logger.WithError(err).Error("Failed to find meetup in DB")
+		return InternalServerError{}
+	}
+
+	session := SessionFromContext(ctx)
+	id := session.Values[UserID]
```

Comment:
Let's name this something easier to understand. (Also assert its type here instead of in other places.)

```suggestion
	userID := session.Values[UserID].(string)
```